app/controllers/scheduler_controller.rb

```diff
@@ -1,6 +1,53 @@
 class SchedulerController < ApplicationController
+  before_action :authenticate_user!
+  # jump date needs to be set before set_scheduled_meetings,
+  # so that the correct time can be used in the course instance query
+  #
+  # The jump date should also be set for JS requests, so that the
+  # source for the autocomplete can be set to the right date.
+  before_action :set_jump_date, only: [:index], :if => [:semester_request?, -> { json_request? || js_request? }]
+  before_action :set_scheduled_meetings, only: [:index], :if => :json_request?
+  before_action :set_enrollment, only: [:create]
 
   # GET /scheduler
+  # GET /scheduler.json
+  # GET /scheduler.js
   def index
   end
+
+  # POST /scheduler.js
+  def create
+  end
+
+  private
+
+  def set_enrollment
+    @enrollment = Enrollment.where(user: current_user, course_instance_id: params[:course_instance_id]).first_or_create
+  end
+
+  # Set the course instances to be rendered in the schedule
+  # This is for the JSON feed that fullcalendar requires
+  def set_scheduled_meetings
+    # timecop should be set here to get the correct enrollments in the query
+    @scheduled_meetings = current_user.enrolled_courses.ongoing(@jump_date || Date.today).includes(:meetings, :course).flat_map(&:meetings).flat_map(&:scheduled_meetings)
+  end
+
+  def set_jump_date
+    jump_date_string = Semester::SAFE_JUMP_DATES[params[:semester][:semester].to_sym][params[:semester][:half].to_sym] + " #{params[:semester][:year]}"
+    @jump_date = DateTime.strptime(jump_date_string, Semester::SAFE_JUMP_DATE_STRPTIME_STRING + ' %Y')
+  end
```

Comment:
I'm also open to name changes. jump_date is no longer a good name because the logic of what I was doing has changed. Open to suggestions.